doc/style-guide-markdown.md

```diff
@@ -0,0 +1,42 @@
+Style Guide for Markdown
+========================
+
+This extends the [general programming](style-guide-programming.md) portion of the [Style Guide](style-guide.md).
+
+
+* Markdown Formatting
+    * Follow GitHub Flavored Markup.
+    * Paragraphs should be all on one long line.
+    * Avoid HTML markup.
+    * Prefer to use indentation instead of code fences.
```

Comment:
I think we chatted about this at one point but I forgot the reason for it.  Readability when editing?  I ask because I find indentation doesn't work as well as code fences especially for other gitlab (must have a space at the end of a line if in a list?).